api/kyverno/v1/rule_types.go

```diff
@@ -10,8 +10,58 @@ import (
 	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type PolicyExceptionRequestSpec struct {
+	// PolicyName is a label to identify the target policy.
+	PolicyName string `json:"policyName" yaml:"policyName"`
+  // RuleName is a label to identify the target rule. 
+	RuleName string `json:"ruleName" yaml:"ruleName"`
+	// Reason explains why the exception exists.
+	Reason string `json:"reason" yaml:"reason"`
+	// ExcludeResources defines when this policy rule should not be applied. The exclude
+	// criteria can include resource information (e.g. kind, name, namespace, labels)
+	// and admission review request information like the name or role.
+	ExcludeResources MatchResources `json:"exclude" yaml:"exclude"`
+}
+
+type PolicyExceptionRequestStatus struct {
+	// State defines the request state. This includes 'PendingApproval' and 'Approved'.
+	State string `json:"state" yaml:"state"`
+}
+
+type PolicyExceptionRequest struct {
+	metav1.TypeMeta   `json:",inline,omitempty" yaml:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// Spec declares policy exception behaviors.
+	Spec PolicyExceptionRequestSpec `json:"spec" yaml:"spec"`
+
+	// Status declares if request is approved.
+	Status PolicyExceptionRequestStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+type ApprovedPolicyException struct {
```

Comment:
Maybe a better name to separate this one from `ApprovedPolicyExceptions`.

Reply:
Maybe change "ApprovedPolicyExceptions" to "Approvals"?

Reply:
Sounds good to me👍